### PR TITLE
Fix build warnings and Add new defconfig 'build_test' to check build warnings and errors

### PR DESCRIPTION
**Pull request by lokeshbv**
- configs/qemu: Add new build config to check compilation errors or warnings
- Fix compilation warnings and errors
- Add "qemu/build_test" build config to travis CI
- kernel/task: Remove references to non-existent CONFIG_MAX_TASK_ARG

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [f5e95bf7d3ac8312a5af1695c6aff0777672e5de] -

**Comment by seinfra** [comment]
Target : [bfc2764097cb19edf03a1f96b7834eec815f45b2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [bfc2764097cb19edf03a1f96b7834eec815f45b2] -

**Comment by seinfra** [comment]
Target : [8bdd0bb5a19a638a0c8125cbe8fbf271d45f9d82] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [8bdd0bb5a19a638a0c8125cbe8fbf271d45f9d82] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [82353d860cf04a6aef8c96930754a591b568eb00] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [729372eb5ec610b35e3d44e05cf8f82cf70bad39] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [729372eb5ec610b35e3d44e05cf8f82cf70bad39] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/network/tc_net_getsockopt.c [hunk 1]
```diff
@@ -165,6 +165,7 @@ static void tc_net_getsockopt_optval_n(int s)
    * @precondition	   :socket file descriptor
    * @postcondition	   :
    */
+#if 0
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/tc_net_getsockopt.c]: We can't know why you commented this. Could you write the reason after ```#if 0```?
```
#if 0 // blah blah
```

File: apps/examples/testcase/le_tc/drivers/tc_loop.c [hunk 2]
```diff
@@ -143,7 +144,7 @@ static void tc_driver_loop_write(void)
 	memset(read_buf, 0, 1024);
 	ret = pnode->u.i_bops->read(pnode, read_buf, 0, 1);
 	TC_ASSERT_GEQ_CLEANUP("loop_read", ret, 0, cleanup(pnode, read_buf));
-	TC_ASSERT_EQ_CLEANUP("loop_write", strncmp(read_buf, "Test loop driver", 16), 0, cleanup(pnode, read_buf));
+	TC_ASSERT_EQ_CLEANUP("loop_write", strncmp((char *)read_buf, "Test loop driver", 16), 0, cleanup(pnode, read_buf));
```
Comment (yashwanth686007) [inline apps/examples/testcase/le_tc/drivers/tc_loop.c]: I think strncmp requires const char * typecasting ?

File: framework/src/wifi_manager/wpa_wifi_utils.c [hunk 3]
```diff
@@ -132,7 +134,7 @@ fetch_scan_results(wifi_utils_scan_list_s **scan_list, slsi_scan_info_t **slsi_s
 					&cur->ap_info.ap_auth_type, &cur->ap_info.ap_crypto_type);
 			strncpy(cur->ap_info.ssid, (char *)wifi_scan_iter->ssid, wifi_scan_iter->ssid_len);
 			cur->ap_info.ssid_length = (unsigned int)wifi_scan_iter->ssid_len;
-			strncpy(cur->ap_info.bssid, (char *)wifi_scan_iter->bssid, WIFI_UTILS_MACADDR_STR_LEN);
+			strncpy((char *)cur->ap_info.bssid, (char *)wifi_scan_iter->bssid, WIFI_UTILS_MACADDR_STR_LEN);
```
Comment (yashwanth686007) [inline framework/src/wifi_manager/wpa_wifi_utils.c]: 2nd argument should be typecasted as const char *??

**Comment by seinfra** [comment]
Target : [374dbdcd831cca24999998adbcac0f4076b9c384] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [374dbdcd831cca24999998adbcac0f4076b9c384] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [31c5631efdf60fdd1a021518b4da6d3eade3c06d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [31c5631efdf60fdd1a021518b4da6d3eade3c06d] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/network/tc_net_getsockopt.c [hunk 4]
```diff
@@ -165,6 +165,7 @@ static void tc_net_getsockopt_optval_n(int s)
    * @precondition	   :socket file descriptor
    * @postcondition	   :
    */
+#if 0 // get has lwip_tcp in the accepticonn option and not in set.
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/tc_net_getsockopt.c]: @pillip8282 Could you review this?

File: build/configs/qemu/build_test/defconfig [hunk 5]
```diff
@@ -0,0 +1,1272 @@
...
+# CONFIG_NETUTILS_TFTPC is not set
+# CONFIG_NETUTILS_TELNETD is not set
+# CONFIG_NETUTILS_SMTP is not set
+# CONFIG_NETUTILS_MQTT is not set
+# CONFIG_NET_SECURITY_TLS is not set
+
+#
+# Wireless
+#
+# CONFIG_WIFI_MANAGER is not set
```
Comment (sunghan-chang) [inline build/configs/qemu/build_test/defconfig]: Please enable this as well.

File: build/configs/qemu/build_test/defconfig [hunk 6]
```diff
@@ -0,0 +1,1272 @@
...
+#
+# Build Debug Options
+#
+CONFIG_DEBUG_SYMBOLS=y
+# CONFIG_FRAME_POINTER is not set
+
+#
+# Logger Module
+#
+# CONFIG_LOGM is not set
```
Comment (sunghan-chang) [inline build/configs/qemu/build_test/defconfig]: please enable.

File: build/configs/qemu/build_test/defconfig [hunk 7]
```diff
@@ -0,0 +1,1272 @@
...
+CONFIG_TASK_MANAGER=y
+CONFIG_TASK_MANAGER_MAX_TASKS=16
+CONFIG_TASK_MANAGER_MAX_MSG=16
+# CONFIG_TASK_MANAGER_USER_SPECIFIC_BROADCAST is not set
+CONFIG_TASK_MANAGER_UNICAST_REPLY_TIMEOUT=0
+
+#
+# Event Loop Framework
+#
+# CONFIG_EVENTLOOP is not set
```
Comment (sunghan-chang) [inline build/configs/qemu/build_test/defconfig]: enable please

File: build/configs/qemu/build_test/defconfig [hunk 8]
```diff
@@ -0,0 +1,1272 @@
...
+
+#
+# Things Management
+#
+# CONFIG_ST_THINGS is not set
```
Comment (sunghan-chang) [inline build/configs/qemu/build_test/defconfig]: enable

**Comment by seinfra** [comment]
Target : [094c3a74808b66d320c7513e00823e38c9ebab6a] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/network/tc_net_getsockopt.c (hunk 4 above)
Comment (lokeshbv) [reply apps/examples/testcase/le_tc/network/tc_net_getsockopt.c]: It's already commented  from being called. check below. This change is just  to commen the unused definition.
https://github.com/Samsung/TizenRT/blob/094c3a74808b66d320c7513e00823e38c9ebab6a/apps/examples/testcase/le_tc/network/tc_net_getsockopt.c#L253

**Pull request merged**